src/command/Commands.ts

```diff
@@ -499,3 +512,338 @@ export class CommandLineSpec implements InputSpec<typeof StringT> {
+            const { app } = getRootStore();
+            if (param.getInt() >= app.diagram.size()) return fail("Index out of range.");
+            return success("");
+          }
+        },
+        {
+          name: "length",
+          paramType: NumberT,
+          description: "the length of the field",
+          check: param => {
+            if (param.isDouble()) return fail("Length must be a integer.");
+            if (param.getInt() <= 0) return fail("Length must be a positive integer.");
+            return success("");
+          }
+        },
+        {
+          name: "name",
+          paramType: StringT,
+          description: "the name of the field"
+        }
+      ]),
+      "Insert a field at the given index"
+    );
+  }
+
+  handle(params: Parameter<ParameterType>[]): HandleResult {
+    this.paramIndex = params[0].getInt();
+    this.paramLength = params[1].getInt();
+    this.paramName = params[2].getString();
+
+    const { app } = getRootStore();
+
+    this.execute();
+
+    let msg: string;
+
+    if (this.paramIndex === 0) msg = 'Inserted field "' + this.paramName + '" to the beginning.';
+    else
+      msg = 'Inserted field "' + this.paramName + '" after "' + app.diagram.getField(this.paramIndex - 1).name + '".';
+
+    return success(msg);
+  }
+
+  execute() {
+    const { app } = getRootStore();
+    app.diagram.insertField(this.paramIndex, new Field(this.paramName, this.paramLength));
+  }
+}
+
+/**
+ * this command is responsible in rearrange the field from a specified index to an another specified index
+ */
+export class MoveCommand extends CancellableCommand {
+  /**
+   * the index of the to-be rearranged field
+   */
+  paramIndex!: number;
+  /**
+   * the new position of the field
+   */
+  paramTargetIndex!: number;
+
+  constructor() {
+    super(
+      "move",
+      buildInputSpecByUsages([
+        {
+          name: "source_index",
+          paramType: NumberT,
+          description: "the index of the source field",
+          check: param => {
+            if (param.isDouble()) return fail("Source index must be a integer.");
+            if (param.getInt() < 0) return fail("Source index must be a positive integer or zero.");
+            const { app } = getRootStore();
+            if (param.getInt() >= app.diagram.size()) return fail("Index out of range.");
+            return success("");
+          }
+        },
+        {
+          name: "destination_index",
+          paramType: NumberT,
+          description: "the index of the destination field",
+          check: param => {
+            if (param.isDouble()) return fail("Destination index must be a integer.");
+            if (param.getInt() < 0) return fail("Destination index must be a positive integer or zero.");
+            const { app } = getRootStore();
+            if (param.getInt() >= app.diagram.size()) return fail("Index out of range.");
```

Comment:
Change to `Destination index out of range.`
Make it clear

Reply:
Updated.